test/test_transforms_tensor.py

```diff
@@ -710,6 +578,156 @@ def test_autoaugment(self):
+        _test_class_op(
+            T.ColorJitter, meth_kwargs=meth_kwargs, test_exact_match=False, device=device,
+            tol=12.1, agg_method="max"
+        )
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('m', ["constant", "edge", "reflect", "symmetric"])
+@pytest.mark.parametrize('mul', [1, -1])
+def test_pad(m, mul, device):
+    fill = 127 if m == "constant" else 0
+
+    # Test functional.pad (PIL and Tensor) with padding as single int
+    _test_functional_op(
+        F.pad, fn_kwargs={"padding": mul * 2, "fill": fill, "padding_mode": m},
+        device=device
+    )
+    # Test functional.pad and transforms.Pad with padding as [int, ]
+    fn_kwargs = meth_kwargs = {"padding": [mul * 2, ], "fill": fill, "padding_mode": m}
+    _test_op(
+        F.pad, T.Pad, device=device, fn_kwargs=fn_kwargs, meth_kwargs=meth_kwargs
+    )
+    # Test functional.pad and transforms.Pad with padding as list
+    fn_kwargs = meth_kwargs = {"padding": [mul * 4, 4], "fill": fill, "padding_mode": m}
+    _test_op(
+        F.pad, T.Pad, device=device, fn_kwargs=fn_kwargs, meth_kwargs=meth_kwargs
+    )
+    # Test functional.pad and transforms.Pad with padding as tuple
+    fn_kwargs = meth_kwargs = {"padding": (mul * 2, 2, 2, mul * 2), "fill": fill, "padding_mode": m}
+    _test_op(
+        F.pad, T.Pad, device=device, fn_kwargs=fn_kwargs, meth_kwargs=meth_kwargs
+    )
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+def test_crop(device):
+    fn_kwargs = {"top": 2, "left": 3, "height": 4, "width": 5}
+    # Test transforms.RandomCrop with size and padding as tuple
+    meth_kwargs = {"size": (4, 5), "padding": (4, 4), "pad_if_needed": True, }
+    _test_op(
+        F.crop, T.RandomCrop, device=device, fn_kwargs=fn_kwargs, meth_kwargs=meth_kwargs
+    )
+
+    # Test transforms.functional.crop including outside the image area
+    fn_kwargs = {"top": -2, "left": 3, "height": 4, "width": 5}  # top
+    _test_functional_op(F.crop, fn_kwargs=fn_kwargs, device=device)
+
+    fn_kwargs = {"top": 1, "left": -3, "height": 4, "width": 5}  # left
+    _test_functional_op(F.crop, fn_kwargs=fn_kwargs, device=device)
+
+    fn_kwargs = {"top": 7, "left": 3, "height": 4, "width": 5}  # bottom
+    _test_functional_op(F.crop, fn_kwargs=fn_kwargs, device=device)
+
+    fn_kwargs = {"top": 3, "left": 8, "height": 4, "width": 5}  # right
+    _test_functional_op(F.crop, fn_kwargs=fn_kwargs, device=device)
+
+    fn_kwargs = {"top": -3, "left": -3, "height": 15, "width": 15}  # all
+    _test_functional_op(F.crop, fn_kwargs=fn_kwargs, device=device)
+
+
+@pytest.mark.parametrize('device', cpu_and_gpu())
+@pytest.mark.parametrize('padding_config', [
+    {"padding_mode": "constant", "fill": 0},
+    {"padding_mode": "constant", "fill": 10},
+    {"padding_mode": "constant", "fill": 20},
+    {"padding_mode": "edge"},
+    {"padding_mode": "reflect"}
+])
+@pytest.mark.parametrize('size', [5, [5, ], [6, 6]])
```

Comment:
This parametrization and the `config["size"] = size` below is a bit unnatural. I think it mostly comes from the fact that we want to avoid passing a default for `fill` but its default is 0 and we already hardcode it here. Maybe parametrizing over 'padding_mode, fill' and 'size' would make more sense, instead of parametrizing over 'padding_config'.

But it's not super important, so let's leave it as is!